rpm/connman.spec

```diff
@@ -218,9 +218,6 @@ cp -a tools/iptables-unit %{buildroot}%{_libdir}/%{name}/tools
 mkdir -p %{buildroot}%{_sysconfdir}/connman/
 cp -a %{SOURCE1} %{buildroot}%{_sysconfdir}/connman/
 
-mkdir -p %{buildroot}/%{_unitdir}/network.target.wants
-ln -s ../connman.service %{buildroot}/%{_unitdir}/network.target.wants/connman.service
```

Comment:
Should there be multi-user.target.wants link still?

Reply:
No idea. The service file has `multi-user.target` in "Before" rule and has always been as "WantedBy" as well. Do we need explicitly set the `multi-user.target` to wait for connmand? Would that make connmand to be started before the user session to allow developer mode login prior to that, i.e., during the security code?

Reply:
`Before` and `After` dependencies are purely for ordering services so they don't do anything to pull any services up.

Anything in `[Install]` section, like this `WantedBy`, only applies to `systemctl enable/disable` and has no effect runtime. In Sailfish OS we've been using packaged symlinks to add wants dependencies when possible so that they stay consistent between devices and are easy to change on upgrades.

Reply:
BTW, I think the early login over USB happens without connman involvement so at least that should be unaffected. IIRC it uses usb-moded and busybox udhcpd.

Reply:
> BTW, I think the early login over USB happens without connman involvement so at least that should be unaffected. IIRC it uses usb-moded and busybox udhcpd.

Yeah it dawned to me as well that the actual legwork happens with other components. But ConnMan does account for the developer mode interface in many ways, the device is set to not be managed and firewall rule is added.

If the mode is set before connmand is started and the rtnl notify is missed that can cause trouble. Maybe that devmode plugin should query the state when started to accommodate to this change.

Reply:
> `Before` and `After` dependencies are purely for ordering services so they don't do anything to pull any services up.
> 
> Anything in `[Install]` section, like this `WantedBy`, only applies to `systemctl enable/disable` and has no effect runtime. In Sailfish OS we've been using packaged symlinks to add wants dependencies when possible so that they stay consistent between devices and are easy to change on upgrades.

So.. the question is do we really need that `multi-user.target.wants`?

Reply:
> So.. the question is do we really need that `multi-user.target.wants`?

I think the question is really which unit should start connman.service. I don't have strong opinion if it must be multi-user.target but certainly there has to be something that starts connman. If upstream is using it, is there any reason not to use it here as well? AFAICT only places currently are connman-vpnd (what is starting that?) and network.target latter of which you are removing now.

Regarding that USB mode, use of rescue mode should be possible already before home is mounted so it's certainly before connman is up.

Reply:
> > So.. the question is do we really need that `multi-user.target.wants`?
> 
> I think the question is really which unit should start connman.service. I don't have strong opinion if it must be multi-user.target but certainly there has to be something that starts connman. If upstream is using it, is there any reason not to use it here as well? AFAICT only places currently are connman-vpnd (what is starting that? Is it D-Bus activation?) and network.target latter of which you are removing now.

Yeah so I guess we are again using something differently here. Instead of relying on the [Install] we do it manually. Yeah, and this kind of things happen then that need to be fixed ;).

> 
> Regarding that USB mode, use of rescue mode should be possible already before home is mounted so it's certainly before connman is up.

I mean that the interface may become managed and depending on the device the interface is of different type, which causes difficulties internally in ConnMan and that is why the devmode plugin was added. There is no hole in the firewall for it either if the device is not detected as it is started way before connmand starts. That is why I think this needs to be rethought a bit to at least make the initial query at startup. Now I just cannot seem to remember how that 2y back created plugin works..

Reply:
> > > So.. the question is do we really need that `multi-user.target.wants`?
> > 
> > 
> > I think the question is really which unit should start connman.service. I don't have strong opinion if it must be multi-user.target but certainly there has to be something that starts connman. If upstream is using it, is there any reason not to use it here as well? AFAICT only places currently are connman-vpnd (what is starting that? Is it D-Bus activation?) and network.target latter of which you are removing now.
> 
> Yeah so I guess we are again using something differently here. Instead of relying on the [Install] we do it manually. Yeah, and this kind of things happen then that need to be fixed ;).

Changed the PR to include adding of `connman{,-vpn}.service` as symlinks to `multi-user.target.wants`. The `network.target` is now according to the specified rules started again after `connmand` but then again, it is kind of expected and with these changes to reduce the initialization there should not be much delay. 

> 
> > Regarding that USB mode, use of rescue mode should be possible already before home is mounted so it's certainly before connman is up.
> 
> I mean that the interface may become managed and depending on the device the interface is of different type, which causes difficulties internally in ConnMan and that is why the devmode plugin was added. There is no hole in the firewall for it either if the device is not detected as it is started way before connmand starts. That is why I think this needs to be rethought a bit to at least make the initial query at startup. Now I just cannot seem to remember how that 2y back created plugin works..

Got my head wrapped around this. Nothing needs to be done. The `rtnl.c` does request the interfaces at startup and through that the devmode interface is detected as the state of `usb-moded` is queried.

Reply:
I think the choice of the target that wants the unit should not be causing delays. It's there to include the service in the startup flow. `After` and `Before` dependencies change the point when the service actually started.